packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -162,17 +165,40 @@ export const App = ({
   // Can be looked at in FXA-7626 or FXA-7184
   return (
     <Router basepath="/">
-      <AuthAndAccountSetupRoutes {...{ isSignedIn }} path="/*" />
-      <SettingsRoutes {...{ isSignedIn }} path="/settings/*" />
+      <AuthAndAccountSetupRoutes {...{ isSignedIn, integration }} path="/*" />
+      <SettingsRoutes {...{ isSignedIn, integration }} path="/settings/*" />
     </Router>
   );
 };
 
 const SettingsRoutes = ({
   isSignedIn,
-}: { isSignedIn: boolean } & RouteComponentProps) => {
+  integration,
+}: { isSignedIn: boolean; integration: Integration } & RouteComponentProps) => {
   const location = useLocation();
-  if (isSignedIn === false) {
+  // TODO: Remove this + config.sendFxAStatusOnSettings check once we confirm this works
+  const config = useConfig();
+
+  // Request and update account data/state to match the browser state for OAuth Sync flows
+  // because: 1) If a user logs in through the browser Sync pairing flow and then accesses
+  // Settings through the browser, they'd have to login again (isSignedIn === false).
+  // 2) If a user disconnects their account through Sync and then logs into another account
+  // using the pairing flow, we hold onto the old session token (isSignedIn === true).
+  const [shouldCheckFxaStatus, setShouldCheckFxaStatus] = useState(
+    isOAuthIntegration(integration) &&
+      integration.isSync() &&
+      config.sendFxAStatusOnSettings
+  );
+  useEffect(() => {
+    (async () => {
+      if (shouldCheckFxaStatus) {
+        await firefox.requestSignedInUser(integration.data.context);
```

Comment:
@vbudhram I think this is fine to do here since Settings doesn't currently get the `integration` prop or `isSignedIn` prop passed to it. I extracted this out into the `firefox` class though since I think it makes sense for that to be our abstraction layer handling this kind of thing, and I'm thinking that's where some of the sync engine bits should be moved to that are currently in the signup container.